core/shared/src/main/scala/scalaz/zio/Managed.scala

```diff
@@ -33,14 +33,27 @@ sealed abstract class Managed[+E, +R] { self =>
 
   final def seq[E1 >: E, R1](ff: Managed[E1, R1]): Managed[E1, (R, R1)] =
     seqWith(ff)((_, _))
+
+  final def parWith[E1 >: E, R1, R2](that: Managed[E1, R1])(f0: (R, R1) => R2): Managed[E1, R2] =
+    new Managed[E1, R2] {
+      override def use[E2 >: E1, A](f: R2 => IO[E2, A]): IO[E2, A] = {
+        val x = self.use(IO.now)
+        val y = that.use(IO.now)
```

Comment:
This is unfortunately incorrect, as you’re extracting the Managed value from its bracketing scope :-/

When `f` runs, `x` and `y` would already have been finalized.

Reply:
@iravid I see, my bad.